File: testdata/fixtures/go-small/internal/store/cache.go
package store

// Cache wraps a Store with hit counting.
type Cache struct {
	backend *Store
	hits    int
}

// NewCache creates a Cache over a fresh Store.
func NewCache() *Cache {
	return &Cache{backend: New()}
}

// Lookup reads through to the backend and counts hits.
func (c *Cache) Lookup(key string) (string, error) {
	v, err := c.backend.Get(key)
	if err == nil {
		c.hits++
	}
	return v, err
}

// Warm preloads the cache with pairs.
func (c *Cache) Warm(pairs map[string]string) {
	for k, v := range pairs {
		c.backend.Set(k, v)
	}
}

File: testdata/fixtures/ts-small/src/cache.ts
import { Store, type Entry, describe } from './store';

/** Read-through cache with hit counting. */
export class Cache {
  private backend = new Store();
  private hits = 0;

  lookup(key: string): Entry | undefined {
    const entry = this.backend.get(key);
    if (entry !== undefined) {
      this.hits += 1;
    }
    return entry;
  }

  warm(pairs: Record<string, string>): void {
    for (const [key, value] of Object.entries(pairs)) {
      this.backend.set(key, value);
    }
  }

  /** Render cache stats for logging. */
  async report(): Promise<string> {
    const greeting = this.lookup('greeting');
    const line = greeting ? describe(greeting) : 'cold';
    return `${line} (hits=${this.hits})`;
  }
}
